### PR TITLE
Show  columns and problematic values instead of whole rows

### PR DESCRIPTION
**Pull request by jsuchome**
Printing whole row looks quite messy. It seems that pointint out to
problematic value might be more user friendly.

This is my current output:
```
Table 'volumes' contains 4 Byte UTF8 characters, which are, incompatible with the 'utf8' encoding used by MariaDB
The following rows are affected.
id=8c9bc2f3-6b04-489f-9d2a-c3b18c9c8b19
         Column:  display_name
         Value:  💩
id=d9fd877b-f75e-438b-8dda-ed7e43038d2a
         Column:  display_description
         Value:  💫
id=f0dfab82-d54d-46f3-97e4-a859b9a4bc00
         Column:  display_name
         Value:  💪
id=4a0b219e-8b86-4352-8551-49aa361c11c2
         Column:  display_name
         Value:  💬
         Column:  display_description
         Value:  💫
```

### CODE REVIEW

File: psql2mysql.py [hunk 1]
```diff
@@ -64,17 +64,18 @@ def _query_utf8mb4_rows(self, table, stringColumns):
 
     def scanTablefor4ByteUtf8Char(self, table):
         stringColumns = self.getStringColumns(table)
-        LOG.info("Scanning Table %s (columns: %s) for problematic UTF8 "
+        LOG.debug("Scanning Table %s (columns: %s) for problematic UTF8 "
                  "characters", table.name, stringColumns)
         rows = self._query_utf8mb4_rows(table, stringColumns)
-        incompatible_rows = []
+        incompatible = []
+        primary = list(table.primary_key)[0].name
```
Comment (rhafer) [inline psql2mysql.py]: Can we include all primary keys here? IIRC at least some tables have use multiple columns for their primary key.

Comment (jsuchome) [reply psql2mysql.py]: Do you happen to know which table to look at?

Comment (jsuchome) [reply psql2mysql.py]: OK, check now

File: psql2mysql.py [hunk 2]
```diff
@@ -176,19 +180,22 @@ def do_prechecks(config):
...
-            print x
-            raise Exception("UTF8 mb4 found")
+            last_id = ""
+            for item in incompatibles:
+                item_id = ','.join(item["primary"])
+                if last_id != item_id:
+                    print item_id
+                last_id = item_id
+                print "\t Column: ", item["column"]
+                print "\t Value: ", item["value"]
```
Comment (rhafer) [inline psql2mysql.py]: I certainly agree with reducing the printed columns here to just small subset. But what was wrong with the table output? (I am also not a huge fan of it, but I think the new format is less appealing)

Are you ok with it I restore the prettytable output (with a reduced list of columns) for now?

File: psql2mysql.py [hunk 3]
```diff
@@ -64,17 +63,22 @@ def _query_utf8mb4_rows(self, table, stringColumns):
...
             for col in stringColumns:
                 if type(row[col]) is not unicode:
                     continue
                 if re.search(ur'[\U00010000-\U0010ffff]', row[col]):
-                    incompatible_rows.append((col, row))
-        return incompatible_rows
+                    incompatible.append({
+                      "column": col,
+                      "value": row[col],
+                      "primary": [ k.name + "=" + row[k.name] for k in primary_keys]
```
[the remark this answers is not in the document]
Comment (jsuchome) [reply psql2mysql.py]: yeah ... and I need to fix the test

**Comment by rhafer** [comment]
@jsuchome Are you looking into the failed unittest? Otherwise I can have a quick look.

File: psql2mysql.py (hunk 2 above)
Comment (jsuchome) [reply psql2mysql.py]: OK, I did not look what prettytable could show. To me it looked like a huge amount of uninteresting texts

Comment (rhafer) [reply psql2mysql.py]: >  To me it looked like a huge amount of uninteresting texts
Yes the original output was too much. I've just push an additional commit. Output looks like this now: (Do you like that better?)

```
+-----------------------------------------+---------------------+-------+
|               Primary Key               |   Affected Column   | Value |
+-----------------------------------------+---------------------+-------+
| id=9dca024e-21ce-46ec-828d-0126fcac0f31 |     display_name    |   💩   |
| id=30dbb809-8706-4a3c-a652-f1c0f671f0a4 | display_description |   💬   |
| id=ae6ff387-75f8-45da-bd5f-ee47acc6d225 |     display_name    |   e💬  |
| id=d293b25b-6ba3-4549-9fc6-5d3ec39d0659 | display_description |   💪   |
+-----------------------------------------+---------------------+-------+
```

Comment (rhafer) [reply psql2mysql.py]: There seems to be a formatting error however with those UTF character. (Hm, how I love it :()

Feel free to remove that commit if you don't like it. It's not that important.

File: psql2mysql.py [hunk 4]
```diff
@@ -176,19 +182,27 @@ def do_prechecks(config):
...
-            x.field_names = ["Affected Column"] + list(table.columns)
-            for row in rows:
-                x.add_row([row[0]] + list(row[1]))
-            print x
-            raise Exception("UTF8 mb4 found")
+            print("The following rows are affected:")
+
+            output_table = PrettyTable()
+            output_table.field_names = [
+                "Primary Key",
```
Comment (jsuchome) [inline psql2mysql.py]: Key(s)

Comment (rhafer) [reply psql2mysql.py]: Hm, sure? Isn't it always a single "Primary Key" made up of multiple columns? </hairsplitting mode=off> :-)

Comment (jsuchome) [reply psql2mysql.py]: Hm, true, that's the we we show it. I think it could be described both ways actually

File: psql2mysql.py (hunk 2 above)
Comment (jsuchome) [reply psql2mysql.py]: This table is really nicer than the one before, I do not really mind that tiny formatting error

**Pull request merged**